envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/groups/NodeMetadata.kt

```diff
@@ -268,6 +272,47 @@ private fun Value.toOutgoingTimeoutPolicy(default: Outgoing.TimeoutPolicy): Outg
     return Outgoing.TimeoutPolicy(idleTimeout ?: default.idleTimeout, requestTimeout ?: default.requestTimeout)
 }
 
+private fun Value.toOAuth(properties: SnapshotProperties): OAuth {
+    val provider = this.field("provider").toOauthProvider(properties)
+    val policy = this.field("policy").toOAuthPolicy()
+    val verification = this.field("verification").toOAuthVerification()
+
+    return OAuth(provider, verification, policy)
+}
+
+fun Value?.toOauthProvider(properties: SnapshotProperties) = this?.stringValue
+    ?.takeIf { it.isNotEmpty() }
+    ?.let { provider ->
+        properties.jwt.providers
+            .map { property -> property.name }
+            .firstOrNull {
+                it == provider
+            } ?: throw NodeMetadataValidationException("Invalid OAuth provider value: $provider")
+    }
+    ?: throw NodeMetadataValidationException("OAuth provider value cannot be null")
+
+fun Value?.toOAuthVerification() = this?.stringValue
+    ?.takeIf { it.isNotEmpty() }
+    ?.let {
+        when (it) {
+            "offline" -> OAuth.Verification.OFFLINE
+            else -> throw NodeMetadataValidationException("Invalid OAuth verification value: $it")
+        }
+     }
+    ?: OAuth.Verification.OFFLINE
```

Comment:
We could make it configurable in properties.

Reply:
What do you mean by configurable? :)

Reply:
Maybe instead setting default in code we should set in properties and use it here